lib/user.rb

```diff
@@ -0,0 +1,26 @@
+class User < ActiveRecord::Base
+  before_create :encrypt
+
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true # TODO: length ?
+
+  has_many :suggested_songs, class_name: "Song", foreign_key: :suggester_id
+
+  has_many :votes
+  has_many :voted_songs, through: :votes, source: :song
+
+  def points
+    results = []
+    suggested_songs.find_each do |song|
+      song.votes.find_each do |vote|
+        results.push vote.value
+      end
+    end
```

Comment:
This is making a lot of queries, and I'd like to do something to cut down on that (joining explicitly, or using `includes`)